internal/provider/resource_gitlab_pages_domain_test.go

```diff
@@ -0,0 +1,124 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabPagesDomain_basic(t *testing.T) {
+	var pagesDomain gitlab.PagesDomain
+	rInt := acctest.RandInt()
+	testAccCheck(t)
```

Comment:
NIT: early exit asap.

```suggestion
	testAccCheck(t)
	
	var pagesDomain gitlab.PagesDomain
	rInt := acctest.RandInt()
```